### PR TITLE
Awesome Books: Plain JavaScript with Objects

### PR DESCRIPTION
**Pull request by popoolahussein**
# Description:

### General Requirements
* Ensured that there are no linter errors.
* Followed correct GitHub Flow.
* Documented work in a professional manner.
* JavaScript Requirements
* Followed best practices for JavaScript.
## Project Requirements
* Basic UI Implementation:
  * Created a simple UI using plain HTML.
* Book Collection Management:
  * Implemented a collection that keeps a list of books using an array of objects.
  * Created a function to add a new book to the collection with title and author.
  * Created a function to remove a book from the collection using the array filter() method.
  *  Displayed all books saved in the collection in the top part of the page. 

## Check lists:
  * Adding a Book:
    * When a user clicks the "Add" button:
      - [x]  A new book is added to the collection.
      - [x] The new book is displayed on the page.

  * Removing a Book:
     * When a user clicks the "Remove" button:
        - [x] The correct book is removed from the collection.
        - [x] The correct book disappears from the page.

   * Data Persistence:
      * Ensured that data is preserved in the browser's memory using localStorage.

## How to Test
  * Clone the repository and navigate to the project directory.
  * Open index.html in a web browser.
  * Add a book using the form and verify it appears in the list.
  * Remove a book using the "Remove" button and verify it disappears from the list.
  * Refresh the page and ensure the books persist using localStorage.

## Screenshot:

Screenshot (391)

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -3,17 +3,28 @@
   <head>
     <meta charset="UTF-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
-    <meta name="Author" content="Hussein" />
+    <meta name="author" content="Hussein" />
```
Comment (adejam) [inline index.html]: You can use you fullname here

File: src/js/main.js [hunk 2]
```diff
@@ -1,6 +1,75 @@
...
-addBooks();
+/**
+ * ? BookCollection object to manage the collection of books
+ * */
+const bookCollection = {
+  books: JSON.parse(localStorage.getItem('books')) || [],
+
+  addBook(book) {
+    this.books.push(book);
+    this.updateLocalStorage();
```
Comment (adejam) [inline src/js/main.js]: why are we using `this`?. we should do our best to use the exact variable names for more clarity

**Review by adejam — CHANGES_REQUESTED** [review]

**Review by adejam — COMMENTED** [review]
Noticed the excessive use of `this` keyword which is mostly not needed especially when in functional programming.

File: index.html (hunk 1 above)
Comment (popoolahussein) [reply index.html]: ##### This comment has been taken care of.

File: src/js/main.js (hunk 2 above)
Comment (popoolahussein) [reply src/js/main.js]: #### This comment is highly appreciated,
#### it has been corrected.

**Pull request merged**